test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -154,6 +159,44 @@ public void testLogReplicationFSMTransitions() throws Exception {
         transition(LogReplicationEventType.REPLICATION_STOP, LogReplicationStateType.INITIALIZED, true);
     }
 
+    /**
+     * Verify the lastSyncType flag is being initialized properly.
+     *
+     */
+    @Test
+    public void testFullSyncStatusOnInit() throws Exception {
+        initLogReplicationFSM(ReaderImplementation.EMPTY);
+
+        final Table<LogReplicationSession, LogReplicationMetadata.ReplicationStatus, Message> statusTable =
+                this.corfuStore.getTable(NAMESPACE, REPLICATION_STATUS_TABLE_NAME);
+
+        // Initial state: Initialized
+        LogReplicationState initState = fsm.getState();
+        assertThat(initState.getType()).isEqualTo(LogReplicationStateType.INITIALIZED);
+
+        transitionAvailable.acquire();
+
+        // Transition #1: Replication Stop (without any replication having started)
+        transition(LogReplicationEventType.REPLICATION_STOP, LogReplicationStateType.INITIALIZED);
+
+        // Default sync value is null so replication status should not be set at all.
+        final int expectedStatusTableLength = 0;
+        Assert.assertEquals(expectedStatusTableLength, statusTable.count());
+
+        // Transition #2: SNAPSHOT Sync Start
+        transition(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST, LogReplicationStateType.IN_SNAPSHOT_SYNC);
+
+        // Sync value should now reflect change to SNAPSHOT sync.
+        Assert.assertEquals(LogReplicationMetadata.SyncType.SNAPSHOT,
+                statusTable.entryStream()
+                        .findAny()
+                        .get()
+                        .getPayload()
+                        .getSourceStatus()
+                        .getReplicationInfo()
+                        .getSyncType());
```

Comment:
What about the snapshotSyncInfo part? are we validating it? cause we don't have info from that part, so I believe this is still sort of inconsistent, cause it will say snapshot ONGOING, with no snapshot sync info available.

Reply:
It does say snapshot ONGOING however, in this test it appears this value is coming from the onEntry function of the "to" state so the test itself is not accurate and will need to be reworked. But the problem still exists as we discussed above.